setup.py

```diff
@@ -118,5 +98,5 @@ def _is_test_pypi():
     extras_require=extras_require,
     entry_points={'console_scripts': ['pyhf=pyhf.cli:cli']},
     dependency_links=[],
-    use_scm_version=_is_test_pypi(),
+    use_scm_version=lambda: {'local_scheme': lambda version: ''},
```

Comment:
Can you add a short paragraph or some links or code snippet to the PR body that outline why the `IS_COMMIT_TAGGED` environment var isn't needed anymore?

Reply:
I don't think it was ever needed... This is the part I'm really confused about to be honest. I tried with just this, without needing the IS_COMMIT_TAGGED and we properly tag on releases, and properly add devN on non-releases.

Reply:
Cool! That's great news.